modules/gdprEnforcement.js

```diff
@@ -47,58 +47,79 @@ const analyticsBlocked = [];
+
+    // For USER ID Module, we pass the submodule object itself as the "module" parameter, this check is required to grab the module code
+    const moduleCode = typeof module === 'string' ? module : module.name;
+
+    // Return GVL ID from user defined gvlMapping
+    if (gvlMapping && gvlMapping[moduleCode]) {
+      gvlid = gvlMapping[moduleCode];
+      return gvlid;
+    }
+
+    gvlid = internal.getGvlidForBidAdapter(moduleCode) || internal.getGvlidForUserIdModule(module) || internal.getGvlidForAnalyticsAdapter(moduleCode);
+  }
+  return gvlid;
+}
+
+/**
+ * Returns GVL ID for a bid adapter. If the adapter does not have an associated GVL ID, it returns 'undefined'.
+ * @param  {string=} bidderCode - The 'code' property of the Bidder spec.
+ * @return {number} GVL ID
+ */
+function getGvlidForBidAdapter(bidderCode) {
+  let gvlid = null;
   bidderCode = bidderCode || config.getCurrentBidder();
   if (bidderCode) {
-    const gvlMapping = config.getConfig('gvlMapping');
-    if (gvlMapping && gvlMapping[bidderCode]) {
-      gvlid = gvlMapping[bidderCode];
-    } else {
-      const bidder = adapterManager.getBidAdapter(bidderCode);
-      if (bidder && bidder.getSpec) {
-        gvlid = bidder.getSpec().gvlid;
-      }
+    const bidder = adapterManager.getBidAdapter(bidderCode);
+    if (bidder && bidder.getSpec) {
+      gvlid = bidder.getSpec().gvlid;
     }
   }
   return gvlid;
 }
 
 /**
- * Returns gvlId for userId module. If a userId modules does not have an associated gvlId, it returns 'undefined'.
+ * Returns GVL ID for an userId submodule. If an userId submodules does not have an associated GVL ID, it returns 'undefined'.
  * @param {Object} userIdModule
- * @retuns {number} gvlId
+ * @return {number} GVL ID
  */
 function getGvlidForUserIdModule(userIdModule) {
-  let gvlId;
-  const gvlMapping = config.getConfig('gvlMapping');
-  if (gvlMapping && gvlMapping[userIdModule.name]) {
-    gvlId = gvlMapping[userIdModule.name];
-  } else {
-    gvlId = userIdModule.gvlid;
-  }
-  return gvlId;
+  return (typeof userIdModule === 'object' ? userIdModule.gvlid : null);
 }
 
 /**
- * Returns gvlId for analytics adapters. If a analytics adapter does not have an associated gvlId, it returns 'undefined'.
+ * Returns GVL ID for an analytics adapter. If an analytics adapter does not have an associated GVL ID, it returns 'undefined'.
  * @param {string} code - 'provider' property on the analytics adapter config
- * @returns {number} gvlId
+ * @return {number} GVL ID
  */
 function getGvlidForAnalyticsAdapter(code) {
-  let gvlId;
-  const gvlMapping = config.getConfig('gvlMapping');
-  if (gvlMapping && gvlMapping[code]) {
-    gvlId = gvlMapping[code];
-  } else {
-    gvlId = adapterManager.getAnalyticsAdapter(code).gvlid;
+  let gvlid = null;
+  if (adapterManager.getAnalyticsAdapter(code) && adapterManager.getAnalyticsAdapter(code).gvlid) {
+    return adapterManager.getAnalyticsAdapter(code).gvlid
   }
-  return gvlId;
+  return gvlid;
```

Comment:
could this also be shortened to just one-line?

```
return adapterManager.getAnalyticsAdapter(code) && adapterManager.getAnalyticsAdapter(code).gvlid || null;
```

Reply:
Made the change. Was reluctant earlier because if the condition before `&&` returns false, the function will return `undefined` and not `null`. Kind of a minor inconsistency, but I don't think that'll play a big role.